src/Text/Pandoc/Writers/Typst.hs

```diff
@@ -88,6 +88,52 @@ pandocToTypst options (Pandoc meta blocks) = do
        Nothing  -> main
        Just tpl -> renderTemplate tpl context
 
+pickTypstAttrs :: [(Text, Text)] -> ([(Text, Text)],[(Text, Text)])
+pickTypstAttrs =
+  partition (not . T.isPrefixOf "typst:text:" . fst) . filter (T.isPrefixOf "typst:" . fst)
+
+formatTypstProps :: [(Text, Text)] -> [Text]
+formatTypstProps =
+  map (\(k,v) -> last (T.splitOn (T.pack ":") k) <> ": " <> v)
```

Comment:
Avoid `last`, as it's a partial function.
Also you don't need `T.pack ":"`; `":"` will work by itself because we have OverloadedStrings.

Reply:
Good points. I had to improve validation to drop keys like `typst:foo:bar`, so I took the opportunity to pull the last element at input time and not worry about invalid input during output.